src/Component/src/Doctrine/Persistence/InMemoryRepository.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Doctrine\Persistence;
+
+use ArrayObject;
```

Comment:
According to symfony coding style, global classes\exceptions shouldn't be imported and used directly.

```suggestion
```